Clarify call-by-value (#12324)
mdn/content: files/en-us/web/javascript/guide/functions/index.md

return number * number;
```

> Parameters are passed to functions **by value**; the value is passed to the function, but if the function then changes the value of the parameter, **the change is not reflected globally or in the caller**.

Reviewer: ```suggestion
Parameters are passed to functions **by value**; the value is passed to the function, but if the function then changes the value of the parameter, **the change is not reflected globally or in the calling code**.
```

> Parameters are passed to functions **by value**; the value is passed to the function, but if the function then changes the value of the parameter, **the change is not reflected globally or in the calling code**.

Reviewer: ```suggestion
Parameters are passed to functions **by value** — so if the code within the body of a function assigns a completely new value to a parameter that was passed to the function, **the change is not reflected globally or in the code which called that function**.
```

> Parameters (such as a number or an object reference) are passed to functions **by value**; the value is passed to the function, but if the function changes the value of the parameter, **this change is not reflected globally or in the calling function**.

Reviewer: ```suggestion
Parameters are passed to functions **by value**; the value is passed to the function, but if the function then changes the value of the parameter, **the change is not reflected globally or in the caller**.
```

> However, when you pass an object as a parameter, if the function changes the object's properties, that change is visible outside the function, as shown in the following example:

Author: Nope, "However" is spurious juxtaposition. Both statements are true, full stop; the second statement does not refine the first.

Reviewer: > Nope, "However" is spurious juxtaposition. Both statements are true, full stop; the second statement does not refine the first.

OK, thanks — I’ve now fixed that over in https://github.com/mdn/content/pull/12344

> If as a parameter you pass a reference — which will necessarily be a reference to an object of some kind — and the function changes the object's properties, that change is visible outside the function, as shown in the following example:

Reviewer: ```suggestion
However, when you pass an object as a parameter, if the function changes the object's properties, that change is visible outside the function, as shown in the following example:
```

> If you pass a reference (i.e., to a non-primitive value, such as {{jsxref("Array")}} or a user-defined object) as a parameter and the function changes the object's properties, that change is visible outside the function, as shown in the following example:

Reviewer: ```suggestion
If as a parameter you pass a reference — which will necessarily be a reference to an object of some kind — and the function changes the object's properties, that change is visible outside the function, as shown in the following example:
```
